iceoryx_posh/source/roudi/application/roudi_app.cpp

```diff
@@ -67,37 +65,28 @@ void RouDiApp::registerSigHandler() noexcept
     sigemptyset(&act.sa_mask);
     act.sa_handler = roudiSigHandler;
     act.sa_flags = 0;
-    if (-1 == sigaction(SIGINT, &act, NULL))
+    if (cxx::makeSmartC(sigaction, cxx::ReturnMode::PRE_DEFINED_SUCCESS_CODE, {0}, {}, SIGINT, &act, nullptr)
```

Comment:
I think we need the `and_then`/`or_else` pattern for smart_c. I created an issue for that: https://github.com/eclipse/iceoryx/issues/418 and this could become really clean.